float32 min_distance
int16 min_angle
int16 dir